addons/pos_restaurant/models/pos_session.py

```diff
@@ -3,64 +3,37 @@
 
 from odoo import models, Command, api
 from odoo.tools import convert
-from itertools import groupby
-from odoo.osv.expression import AND
 import json
 
 class PosSession(models.Model):
     _inherit = 'pos.session'
 
-    def _pos_ui_models_to_load(self):
-        result = super()._pos_ui_models_to_load()
-        if self.config_id.module_pos_restaurant:
-            result.append('restaurant.floor')
-        return result
+    def _load_data_params(self, config_id):
+        params = super()._load_data_params(config_id)
 
-    def _loader_params_restaurant_floor(self):
-        return {
-            'search_params': {
+        params.update({
+            'restaurant.floor': {
                 'domain': [('pos_config_ids', '=', self.config_id.id)],
                 'fields': ['name', 'background_color', 'table_ids', 'sequence'],
             },
-        }
-
-    def _loader_params_restaurant_table(self):
-        return {
-            'search_params': {
-                'domain': [('active', '=', True)],
+            'restaurant.table': {
+                'domain': lambda data: [('active', '=', True), ('floor_id', 'in', [floor['id'] for floor in data['restaurant.floor']])],
```

Comment:
lambda domain is not extensible. Perhaps extract each as a method.

Reply:
I think is better to keep it like that, if you need to extends it, just rewrite it.

One method by domain will maybe look messy